packages/functional-tests/pages/login.ts

```diff
@@ -360,10 +357,8 @@ export class LoginPage extends BaseLayout {
   }
 
   async clickDontHaveRecoveryKey() {
-    return Promise.all([
-      this.page.locator(selectors.LINK_LOST_RECOVERY_KEY).click(),
-      this.page.waitForNavigation(),
-    ]);
+    await this.page.locator(selectors.LINK_LOST_RECOVERY_KEY).click(),
```

Comment:
huh is this valid js/ts syntax?

Reply:
API: https://playwright.dev/docs/api/class-locator#locator-click
Impl: https://github.com/microsoft/playwright/blob/b5195122d95edb45463df54fa081505d5957eff2/packages/playwright-core/src/client/locator.ts#L95-L97

Reply:
I mean, unless my eyes are failing at matching parens (very possible!), that's a comma not semicolon following a function call.

Reply:
HA, no great eyes, I totally missed that comma (even with your callout).
I think this [magically] works because it's the [comma operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comma_Operator), although it should be a semicolon:
> _"The comma (`,`) operator evaluates each of its operands (from left to right) and returns the value of the last operand."_

From the looks of this locally, it waits for `a()` to resolve before calling `b()`, which might explain why this works:
```js
main();

async function main() {
  await a(), await b();
}

async function a() {
  await sleep(3_000);
  console.log("a");
}

async function b() {
  console.log("b");
}

async function sleep(ms=2000) {
  return new Promise(resolve => setTimeout(resolve, ms));
}
```

Reply:
I know that semicolons aren't always required but in this case I guess I just didn't read that as two expressions with a comma in between.  (And thanks for humoring me.  I always appreciate these detailed comments.)

Reply:
And ye olde ESLint for this would probably be https://eslint.org/docs/latest/rules/no-sequences
> _"The comma operator includes multiple expressions where only one is expected. It evaluates each operand from left to right and returns the value of the last operand. However, this frequently obscures side effects, and its use is often an accident."_

Reply:
TIL 😂, but this was definately a typo